### DISCUSSION TITLE
Cobigen in a stand-alone GUI or Web Application

### DISCUSSION THREAD

**Discussion by quantumfate** [devonfw/cobigen#1609] [Ideas]
As a Cobigen user I want to be able to generate Code for my project independently of the platform that I am using. As for now the ease of access is strongly limited by the Devon IDE. Even though it's possible to integrate Cobigen into the IntelliJ IDE it takes a large amount of effort and a lot of documentation that additionally has to be maintained. 

This discussion is supposed to be a set of ideas to initiate this project. We need to brainstorm ideas for the following:
- Features
- Web Application? GUI?
- Implementation
- Supported Languages
- Requirements (Java version update)

Specifically wee need to know where we want to go and which stack/technologies will help us.

@devonfw/cobigen_templates

**Comment by quantumfate** [comment]
I want to suggest the [vaadin Framework](https://vaadin.com/). I never used it personally but it is a modern Web Framework based on Java. Therefor I think it could be helpful.

**Comment by Lur1an** [comment]
An idea could be to de-couple the GUI from the cobigen-core project, this would give us complete freedom on which technologies/languages/frameworks to use without worrying about cobigen dependencies.
The GUI could then be build with popular desktop multi-platform frameworks like ElectronJS or JavaFX.
This idea can also be applied to the Eclipse plugin (if it will be maintained), decoupling cobigen from eclipse-plugin would allow us to use more recent java versions for core.
